src/Http/Authentication.Abstractions/src/IAuthenticationService.cs

```diff
@@ -18,7 +18,7 @@ public interface IAuthenticationService
         /// <param name="context">The <see cref="HttpContext"/>.</param>
         /// <param name="scheme">The name of the authentication scheme.</param>
         /// <returns>The result.</returns>
-        Task<AuthenticateResult> AuthenticateAsync(HttpContext context, string scheme);
+        Task<AuthenticateResult?> AuthenticateAsync(HttpContext context, string? scheme);
```

Comment:
Where would AuthenticateAsync return a null value?

Reply:
https://github.com/dotnet/aspnetcore/blob/master/src/Http/Authentication.Core/src/AuthenticationService.cs#L81-L98

Reply:
That seems precautionary, it doesn't look like any of the handlers return null from the call to AuthenticateAsync on line 80.

Reply:
I was walking back the annotation based on the implementation. We could make it non-null if we expect this to be non-null.

Reply:
Yeah, try it. It looks like IAuthenticationHandler.AuthenticateAsync is already non-nullable.